### DISCUSSION TITLE
How can I modify footer, navbar and other personalization Settings on the server after deploying docusaurus

### DISCUSSION THREAD

**Discussion by hexinnnn** [facebook/docusaurus#9114] [Q&A] [answered]
I wanted to change navbar.title and footer into configurable ones, but there was no configuration file after packaging, so it could not be implemented. I could only dynamically add script code and modify title. Only for the first time

**Comment by hexinnnn** [comment]
<img width="733" alt="image" src="https://github.com/facebook/docusaurus/assets/28981962/6559915f-b80f-4c5c-9f09-940eb03bf35b">
These are some of the Settings I want to update in real time on the server without having to build again

**Reply by slorber** [reply]
Docusaurus is a static site generator: by default everything is static and you can't update anything in real-time.

I don't understand why you would even want to update the site title or copyright in real time 🤷‍♂️  what's the use case and business outcome you are trying to achieve with that?

What does "in real time" even mean? If you want you can batch edit/replace all the static html files that have been built.

Otherwise you want "true realtime" you can swizzle the theme React components and customize them: you are free to use any React code, including code that receives events from websocket, server-sent-even or whatever "realtime system" you can think of. That requires frontend development knowledge and that's outside the scope of Docusaurus core, which remains static by default.

**Reply by hexinnnn** [reply]
Thank you for your answer. Our business may occasionally need to change the title, but we don't want to package and build again. I understand

**Reply by hexinnnn** [reply]
After I dynamically modify the generated html static file, it will still be overwritten, should be assets in js also have dynamic operations, what should I  @slorber

**Reply by slorber** [reply, marked as answer]
React hydrates the frontend application once the JS is loaded so the title is also present in the JS file. As suggested, swizzle is the best solution in case you need to customize your Docusaurus site. Anything else (modifying the output html+js) would also work but is more a workaround than a recommendation.

**Reply by hexinnnn** [reply]
thx
